doc/ingest.md

```diff
@@ -86,6 +88,34 @@ The network indexer nodes expect that metadata begins with a `uvarint` identifyi
   * the proposed `uvarint` protocol is `0x3D0000`.
   * the following bytes are not yet defined.
 
+#### ExtendedProvider
+
+The `ExtendedProvider` field allows for specification of provider families, in cases where a provider operates multiple PeerIDs, perhaps with different access methods between them, but over the same database of content.
+
+```
+type ExtendedProvider struct {
+    Providers [Provider]
+    Override bool
+}
+
+type Provider struct {
+    Provider String
+    Addresses [String]
+    Metadata optional Bytes
+    Signature Bytes
+}
+```
+
+* If `Metadata` is not specified for a `Provider`, the metadata from the encapsulating  `Advertisement` will be used instead.
+* If a `Provider` listing is written with no `ContextID`, those peers will be returned for all published advertisements for the advertisement Publisher.
+  * If `Override` is set on an `ExtendedProvider` entry on an advertisement with a `ContextID`, it indicates that any specified chain-level set of providers should not be returned for that context ID.
+  * If `Override` is not set on an entry for an advertisement with a `ContextID`, it will be combined as a union with any set chain-level `ExtndedProviders`.
+  * If `Override` is set on `ExtendedProvider` for an advertisement without a `ContextID`, the entry can be considered invalid and should be ignored.
+* The `Signature` for an `ExtendedProvider` is signed by the key of that extended `Provider`.
+  * The full advertisement object is serialized, with all instances of `Signature` replaces with an empty array of bytes.
+  * This serialization is then hashed, and the hash is then signed.
+  * The provider from the surrounding advertisement must be present as a `Provider` in the `ExtendedProvider` object, and must sign in this way as well. It does not need to provide multiaddrs or transports, but must be present.
```

Comment:
```suggestion
  * The `Provider` from the encapsulating advertisement must be present in the `Providers` of the `ExtendedProvider` object, and must sign in this way as well. It may omit `Metadata` and `Addresses` if they match the values already set at the encapsulating advertisement. However, `Signature` must be present.
```

We used "encapsulating" earlier. Feel free to change all to "surrounding" if that reads better. Consistent terminology is the nice thing to have.